staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -602,9 +654,19 @@ func (e *Store) Update(ctx context.Context, name string, objInfo rest.UpdatedObj
 			return nil, false, err
 		}
 	}
+	success = true // used in the defer'ed cleanup
 	return out, creating, nil
 }
 
+// This is a helper to convert UpdateOptions to CreateOptions for the
+// create-on-update path.
+func newCreateOptionsFromUpdateOptions(in *metav1.UpdateOptions) *metav1.CreateOptions {
+	return &metav1.CreateOptions{
+		DryRun:       in.DryRun,
+		FieldManager: in.FieldManager,
+	}
```

Comment:
I can't remember if populating the type meta is important, but patch does when converting PatchOptions to `{Create,Update}Options`:

https://github.com/kubernetes/kubernetes/blob/0ad06e991ab0d0708593e3e6415c521c6b76b7d1/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go#L647-L671

Reply:
ACK